pkg/apis/cartographer/v1alpha1/workload_helpers.go

```diff
@@ -97,6 +97,33 @@ func (w *WorkloadSpec) MergeServiceAccountName(serviceAccountName string) {
 	w.ServiceAccountName = serviceAccountNamePtr
 }
 
+func (w *Workload) ReplaceMetadata(updates *Workload) {
+	w.ReplaceAnnotations(updates)
```

Comment:
Should something like the following be added? 

```
	w.GenerateName = updates.GenerateName
	w.DeletionGracePeriodSeconds = updates.DeletionGracePeriodSeconds
	w.OwnerReferences = updates.OwnerReferences
	w.Finalizers = updates.Finalizers
	w.ManagedFields = updates.ManagedFields
```

If so, we'd be facing the same problem: if fields that user can manage start to increase, this is going to start very difficult to manage.